assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -493,22 +493,14 @@ export default Component.extend(TextareaTextManipulation, ComposerUploadUppy, {
 
   @action
   internalSendMessage() {
-    if (this._messageIsValid()) {
-      return this.sendMessage(this.value, this.uploads).then(() =>
-        this.reset()
-      );
-    }
+    return this.sendMessage(this.value, this.uploads).then(this.reset);
   },
 
   @action
   internalEditMessage() {
-    if (this._messageIsValid()) {
-      return this.editMessage(
-        this.editingMessage,
-        this.value,
-        this.uploads
-      ).then(() => this.reset());
-    }
+    return this.editMessage(this.editingMessage, this.value, this.uploads).then(
+      this.reset
```

Comment:
Note doing it this way can mess up the `this` context. If it works that's great but it's generally safer to do it via `() => this.reset()`

Reply:
Since it's an action, `this` is bound properly

Reply:
I mean `reset` is an action so it's bound. Do you prefer the other style? I can swap